sim/core/spell_mod.go

```diff
@@ -406,19 +427,27 @@ func removeDamageDoneAdd(mod *SpellMod, spell *Spell) {
 }
 
 func applyPowerCostPercent(mod *SpellMod, spell *Spell) {
-	spell.CostMultiplier += mod.floatValue
+	if spell.Cost != nil {
+		spell.Cost.Multiplier += int32(mod.intValue)
+	}
 }
 
 func removePowerCostPercent(mod *SpellMod, spell *Spell) {
-	spell.CostMultiplier -= mod.floatValue
+	if spell.Cost != nil {
+		spell.Cost.Multiplier -= int32(mod.intValue)
+	}
 }
 
 func applyPowerCostFlat(mod *SpellMod, spell *Spell) {
-	spell.DefaultCast.Cost += mod.floatValue
+	if spell.Cost != nil {
+		spell.Cost.FlatModifier += int32(mod.intValue)
+	}
 }
 
 func removePowerCostFlat(mod *SpellMod, spell *Spell) {
-	spell.DefaultCast.Cost -= mod.floatValue
+	if spell.Cost != nil {
+		spell.Cost.FlatModifier -= int32(mod.intValue)
+	}
 }
 
 func applyRunicPowerCostFlat(mod *SpellMod, spell *Spell) {
```

Comment:
Can't this be removed now and wrapped into `applyPowerCostFlat()` with the appropriate `ResourceType`?

Reply:
Ah yea I actually can, just need to look for the ResourceType 👍

Reply:
I tried, but it can't. 
RunicPower + Runes have a different way of tracking cost and currently this directly modifies the float value of the cost. It never actually uses additive/multiplicative modifers. 
As my intent is not to fully refactor the way DK RP/Runes are spent I rather leave it as is